src/messaging/ExchangeMgr.cpp

```diff
@@ -65,20 +67,27 @@ ExchangeManager::ExchangeManager()
 
 CHIP_ERROR ExchangeManager::Init(SecureSessionMgr * sessionMgr)
 {
+    ExchangeContext * ec = mContextPool;
+
     if (mState != State::kState_NotInitialized)
         return CHIP_ERROR_INCORRECT_STATE;
 
     mSessionMgr = sessionMgr;
 
     mNextExchangeId = GetRandU16();
 
+    for (int i = 0; i < CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS; i++, ec++)
+        ec = nullptr;
```

Comment:
I don't understand what this is trying to do.  It keeps changing the value of `ec`, but doesn't do anything with the pointer... If it's trying to "clear out" the `ExchangeContext`s in `mContextPool`, this is not going to do it.